tls/s2n_config.c

```diff
@@ -808,6 +812,30 @@ int s2n_config_set_extension_data(struct s2n_config *config, s2n_tls_extension_t
     return 0;
 }
 
+int s2n_config_set_custom_critical_oids(struct s2n_config *config, const char *const *custom_critical_oids, uint32_t custom_oid_count)
+{
+    POSIX_ENSURE_REF(config);
```

Comment:
nit: `POSIX_ENSURE_REF` will return `S2N_ERR_NULL`, which is an internal error:
https://github.com/aws/s2n-tls/blob/92f7827c8487eb2a99b443aec6ee7d1df031b1bf/utils/s2n_safety_macros.h#L310

However, if the user provides a null argument, that's a usage error. It'd be more accurate to throw `S2N_ERR_INVALID_ARGUMENT` here instead.